src/librustc/session/config.rs

```diff
@@ -720,6 +750,27 @@ pub fn build_configuration(sess: &Session) -> ast::CrateConfig {
     }
     let mut v = user_cfg.into_iter().collect::<Vec<_>>();
     v.extend_from_slice(&default_cfg[..]);
+
+    // Before we touch LLVM, make sure that multithreading is enabled.
+    unsafe {
```

Comment:
You mean in `librustc_driver`? AFAICT it is not possible to invoke functions in `librustc_driver` from `librustc` because the crates would have a circular dependency. Should I also try to move `build_configuration` to another crate? (is it possible/does it make sense?)